src/User/Infra/Doctrine/ObjectFieldMappings.php

```diff
@@ -24,7 +24,14 @@
         Features\TokenCredential::class => Credential\Token::class,
     ];
 
-    public static function provideObjectFieldMappings(): iterable
+    private $keyMaxLength;
+
+    public function __construct(int $keyMaxLength = 255)
+    {
+        $this->keyMaxLength = $keyMaxLength;
+    }
+
+    public static function provideObjectFieldMappings(int $keyMaxLength = 255): iterable
```

Comment:
so we update the contract to `ObjectFieldMappingsProviderInterface::provideObjectFieldMappings(MappingConfig $config)`

i was also thinking to rename this part to simply `MappingProviderInterface`